lib/logstash/codecs/identity_map_codec.rb

```diff
@@ -182,11 +202,28 @@ def flush(&block)
 
   def close()
     cleaner.stop
+    auto_flusher.stop
     all_codecs.each(&:close)
   end
   # end Codec API
   # ==============================================
 
+  def auto_flush_mapped
+    nowf = Time.now.to_f
+    identity_map.each do |identity, compo|
+      next unless nowf > compo.auto_flush_timeout
+      compo.codec.auto_flush
```

Comment:
Small question any specific reason why we use the float value of the Time object? Faster comparison?